docs/v1/P4Runtime-Spec.mdk

```diff
@@ -1017,6 +1019,129 @@ table my_ipv4_lkup {
 }
 ~ End p4example
 
+### Custom Annotations in P4Info
+P4 allows flexible annotations[@P4Annotations]. Users may add their own
+annotations as long as they obey the language syntax. To facilitate consumption
+of user-defined annotations, P4Info has an `AnnotationMap` for any element which
+can accept such annotations, &ie; the `PkgInfo`, `Preamble`, `MatchField`,
+`ActionRef`, `Action` and `ControllerPacketMetadata` messages, as well as the
+type specification messages in `p4types.proto`. The `AnnotationMap` contains a
+suitable parsed representation for the different types of annotation bodies,
+namely: `expressions` and `properties`. Empty annotation bodies are signified by
+an empty `value` field  in the `AnnotationMap` entry. 
+
+Note that `AnnotationBody` allows for simultaneous `expressions` and
+`properties` entries even though the P6-16 specification grammar does not at this
+writing.
+
+The `AnnotationBody` and `AnnotationMap` messages are defined in `p4types.proto`:
+
+~ Begin Proto
+// A parsed representation of the annotation body.
+// Ex. empty annotation:           @my_anno1()
+// Ex. expression list annotation: @my_anno2(1+1, "Hello", 1 << 3)
+// Ex. KV-pair annotation:         @my_anno3(a=b,c=d)
+// Ex. Cumulative kv-pair          @my_anno3(e=f)  // adds to my_anno3's properties
```

Comment:
I took a look at the rendered HTML (https://s3-us-west-2.amazonaws.com/p4runtime/travis/chris/add-anno-property-maps/P4Runtime-Spec.html#sec-custom-annotations-in-p4info) and this line was too long for the text box